presto-password-authenticators/src/main/java/io/prestosql/plugin/password/filebasedauthenticator/Users.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.password.filebasedauthenticator;
+
+import com.google.common.collect.ImmutableSet;
+
+import javax.naming.AuthenticationException;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Set;
+
+public class Users
+{
+    private final Set<User> users;
+
+    public Users(Set<User> users)
+    {
+        this.users = users;
+    }
+
+    public Set<User> getUsers()
+    {
+        return users;
+    }
+
+    public void validateUser(User user)
+            throws AuthenticationException
+    {
+        if (!users.contains(user)) {
+            throw new AuthenticationException("Authentication failed");
+        }
+    }
+
+    public static Users readUsers(String filename)
+            throws IOException
+    {
+        ImmutableSet.Builder users = ImmutableSet.builder();
```

Comment:
This could be simplified using the `Files.lines()` method which returns a `Stream`